nvflare/apis/impl/wf_comm_client.py

```diff
@@ -92,27 +96,42 @@ def broadcast_and_wait(
             task.client_tasks.append(client_task)
             task.last_client_task_map[client_task.id] = client_task
 
-            # task_cb_error = self._call_task_cb(task.before_task_sent_cb, client, task, fl_ctx)
-            # if task_cb_error:
-            #     return self._make_error_reply(ReturnCode.ERROR, targets)
+            task_cb_error = self._call_task_cb(task.before_task_sent_cb, client, task, fl_ctx)
+            if task_cb_error:
+                return self._make_error_reply(ReturnCode.ERROR, targets)
+
+        if task.timeout < 0:
+            raise ValueError(f"The task timeout must >= 0. But got {task.timeout}")
 
-        if task.timeout <= 0:
-            raise ValueError(f"The task timeout must > 0. But got {task.timeout}")
+        if task.timeout == 0:
+            task.timeout = MAX_TASK_TIMEOUT
```

Comment:
This might not be clear to the users, we need to document this somewhere saying set to 0 will be MAX_TASK_TIMEOUT

Reply:
I can add this to the `class Task` docstring https://github.com/NVIDIA/NVFlare/blob/main/nvflare/apis/controller_spec.py#L91